### PR TITLE
[AI] Remove unused code

### PR DESCRIPTION
**Pull request by Cjkjvfnby**
Remove code that is not used.   Most of it is related to old research code that was recently removed.

### CODE REVIEW

File: default/python/AI/AIDependencies.py [hunk 1]
```diff
@@ -42,12 +42,9 @@
 # <editor-fold desc="Basic Game Rules">
 # TODO look into (enabling) simply retrieving the below values via UserString
 INDUSTRY_PER_POP = 0.2
-RESEARCH_PER_POP = 0.2
```
Comment (Grummel7) [inline default/python/AI/AIDependencies.py]: This shouldn't be unused, there is actually a typo (or a c&p error) in colonization.calculate_research.

File: default/python/AI/AIDependencies.py [hunk 2]
```diff
@@ -318,11 +305,6 @@ def not_affect_by_special(special: str, species: str) -> bool:
 }
 # </editor-fold>
 
-# <editor-fold desc="Detection related specials">
-PANOPTICON_SPECIAL = "PANOPTICON_SPECIAL"
-PANOPTICON_DETECTION_BONUS = 10
```
Comment (Grummel7) [inline default/python/AI/AIDependencies.py:322]: Looks like another special not evaluated by colonization rating. Please keep it. The bonus should probably become a named value.

File: default/python/AI/AIDependencies.py [hunk 3]
```diff
@@ -349,21 +331,13 @@ def not_affect_by_special(special: str, species: str) -> bool:
 NEST_DOMESTICATION_TECH = "SHP_DOMESTIC_MONSTER"
 
 LRN_ARTIF_MINDS_1 = "LRN_NASCENT_AI"
-LRN_ARTIF_MINDS_2 = "LRN_NASCENT_AI"
 LRN_ALGO_ELEGANCE = "LRN_ALGO_ELEGANCE"
 GRO_PLANET_ECOL = "GRO_PLANET_ECOL"
-GRO_SUBTER_HAB = "GRO_SUBTER_HAB"
-LRN_PHYS_BRAIN = "LRN_PHYS_BRAIN"
```
Comment (Grummel7) [inline default/python/AI/AIDependencies.py:354]: This values are likely redundantly defined in the tech lists. I guess I'll sort it out when working on tech, for now they can go.

File: default/python/AI/EnumsAI.py [hunk 4]
```diff
@@ -18,13 +13,6 @@ class PriorityType(IntEnum):
     PRODUCTION_INVASION = 9
     PRODUCTION_MILITARY = 10
     PRODUCTION_BUILDINGS = 11
-    RESEARCH_LEARNING = 12
-    RESEARCH_GROWTH = 13
-    RESEARCH_PRODUCTION = 14
-    RESEARCH_CONSTRUCTION = 15
-    RESEARCH_ECONOMICS = 16
-    RESEARCH_SHIPS = 17
-    RESEARCH_DEFENSE = 18
```
Comment (Grummel7) [inline default/python/AI/EnumsAI.py:27]: I think I got problems loading old safe files after removing values here.

Comment (Cjkjvfnby) [reply default/python/AI/EnumsAI.py:27]: They are supposed to be unused, so they should not be in the saves.  Will take a look.

File: default/python/AI/AIDependencies.py (hunk 2 above)
Comment (Cjkjvfnby) [reply default/python/AI/AIDependencies.py:322]: Keeping this in code want help.  Maybe we should put it to the issue?  I guess there might be other species that missed.

File: default/python/AI/AIDependencies.py (hunk 1 above)
Comment (Cjkjvfnby) [reply default/python/AI/AIDependencies.py]: I'll check and fix it in separate PR.

Comment (Cjkjvfnby) [reply default/python/AI/AIDependencies.py]: https://github.com/freeorion/freeorion/pull/4245

File: default/python/AI/EnumsAI.py (hunk 4 above)
Comment (Cjkjvfnby) [reply default/python/AI/EnumsAI.py:27]: I have 1 old save from August 2021 it works for me both with master and this branch.

**Comment by Cjkjvfnby** [comment]
@Grummel7 release is over, so it's time to get bach to this thing.  Could you take a look?  (no time pressure, I'll be off my computer this weekend)

**Pull request merged**